core/chains/evm/forwarders/orm.go

```diff
@@ -66,3 +69,47 @@ func (o *orm) FindForwarders(offset, limit int) (fwds []Forwarder, count int, er
 	}
 	return
 }
+
+// FindForwardersByChain returns all forwarder addresses for a chain.
+func (o *orm) FindForwardersByChain(evmChainId utils.Big) (fwds []Forwarder, err error) {
+	sql := `SELECT * FROM evm_forwarders where evm_chain_id = $1 ORDER BY created_at DESC, id DESC`
+	if err = o.q.Select(&fwds, sql, evmChainId); err != nil {
+		return
+	}
+	return
+}
+
+func (o *orm) FindForwardersInListByChain(evmChainId utils.Big, addrs []common.Address) ([]Forwarder, error) {
+	var fwdrs []Forwarder
+
+	arg := map[string]interface{}{
+		"addresses": addrs,
+		"chainid":   evmChainId,
+	}
+
+	query, args, err := sqlx.Named(`
+		SELECT * FROM evm_forwarders 
+		WHERE evm_chain_id = :chainid
+		AND address IN (:addresses)
+		ORDER BY created_at DESC, id DESC`,
+		arg,
+	)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to format query")
+	}
+
+	query, args, err = sqlx.In(query, args...)
+	if err != nil {
+		return nil, err
```

Comment:
wrap err pls.

All calls to external libs that return error should wrap at the earliest possible opportunity so we can find where they came from and add stack information

Reply:
Done here and other spots